grails-app/controllers/org/pih/warehouse/api/InventoryItemApiController.groovy

```diff
@@ -0,0 +1,30 @@
+/**
+* Copyright (c) 2012 Partners In Health.  All rights reserved.
+* The use and distribution terms for this software are covered by the
+* Eclipse Public License 1.0 (http://opensource.org/licenses/eclipse-1.0.php)
+* which can be found in the file epl-v10.html at the root of this distribution.
+* By using this software in any fashion, you are agreeing to be bound by
+* the terms of this license.
+* You must not remove this notice, or any other, from this software.
+**/
+package org.pih.warehouse.api
+
+import org.codehaus.groovy.grails.web.json.JSONObject
+import org.pih.warehouse.inventory.InventoryService
+
+class InventoryItemApiController {
+
+    InventoryService inventoryService
+
+    def validateLot = {
+        JSONObject jsonObject = request.JSON
+        Boolean areLotsValid = inventoryService.areLotsValid(jsonObject?.items)
```

Comment:
Let's just make this an action "validateInventoryItems" or "validateProductLots" instead of create a new logic operator. Also let's call product availability service since we can implement the logic there without needing to add the circular dependency between inventory service and product availability service.
```
Boolean isValid = productAvailabilityService.validateInventoryItems(jsonObject.inventoryItems)
```

Reply:
Why are we passing multiple items?